server/consumer.go

```diff
@@ -4821,16 +4823,16 @@ func (o *consumer) deliverMsg(dsubj, ackReply string, pmsg *jsPubMsg, dc uint64,
 	// Update delivered first.
 	o.updateDelivered(dseq, seq, dc, ts)
 
-	// Send message.
-	o.outq.send(pmsg)
```

Comment:
Maybe we could send a message to the client before `o.trackPending` captures the sequence below, but it likely doesn't happen in practice. Moving it to be below `o.trackPending` doesn't hurt.